fbw-a32nx/src/systems/fmgc/src/guidance/vnav/VerticalProfileManager.ts

```diff
@@ -761,6 +761,14 @@ export class VerticalProfileManager {
 
     return this.flightPlanService.active?.hasTooSteepPathAhead();
   }
+
+  shouldShowLatDiscontinuityAhead(): boolean {
+    return (
+      this.fcuModes.isInNavMode() &&
+      this.flightPlanService.active?.hasDiscontinuityLegNext() &&
+      this.mcduProfile.waypointPredictions.get(this.flightPlanService.activeLegIndex)?.secondsFromPresent < 30
```

Comment:
It seems like what you really want is the last leg before a discont, with some logic to determine which segments to consider..